core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1814,4 +1814,8 @@ package object config {
     .bytesConf(ByteUnit.BYTE)
     .createOptional
 
+  private[spark] val RESOURCE_PROFILE_MERGE_CONFLICTS =
+    ConfigBuilder("spark.scheduler.resourceProfile.mergeConflicts")
+      .booleanConf
```

Comment:
Hi, @tgravescs .
Could you add `.doc` and `.version` like the other cons?

Reply:
yes, I will add

Reply:
When I look through confs which introduced around the feature, I feel that they're kind of messy. For example, we already have:

- spark.resources.discoveryPlugin
- spark.driver.resourcesFile
- spark.driver.resource.GPU.amount

Since they're all from the same feature, so I'm thinking that if we can make them under the same namespace, e.g. `spark.resource`.

WDYT?

Reply:
our normal format for differentiating driver and executor configs is spark.driver. and spark.executor., so I am not in favor or changing that to be spark.resource.driver spark.resource.executor as I think that is not consistent.
I'm fine with changing to spark.driver.resource. and spark.executor.resource. or if not specific spark.resource.
I'm not sure what you would call spark.driver.resourcesFile that makes sense though, unless you call it. spark.driver.resource.resourcesFile which seems like just extra but is consistent among the feature.

Similar here, spark.scheduler is a prefix that tells you what component is affects, if we want to call this spark.scheduler.resource.resourceProfileMergeConflicts or something like that I'd be ok with that.

Reply:
Yeah, I also know the problem here. And I just wonder if they would not be such friendly for user. We shall doc them in a well organized way at least.